examples/async-reply-hook/src/wasm.rs

```diff
@@ -0,0 +1,34 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use gstd::{critical, debug, exec, msg, prelude::*, ActorId};
+
+#[gstd::async_main]
+async fn main() {
```

Comment:
I'd patch demo and test to send 2 messages, but apply handle reply hook only for one of them to be able to assert that they're properly called one for each, rather than just always running

Reply:
take a look at the test now